go/src/koding/socialapi/presence/presence.go

```diff
@@ -0,0 +1,38 @@
+package presence
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+
+	"koding/socialapi"
+)
+
+type Client struct {
```

Comment:
ok i see the problem, you dont have access to `clientId` since it is in the transport cache/auther.

on the other hand NewRequest is exported https://github.com/koding/koding/pull/9863/files#diff-9da7fe549db0224bc2de4a603ce84d51R72 so you have acccess to the http.request itself